stackrox/acs-fleet-manager
# ROX-15155: Add ACS Operator install/upgrade from Fleetshard-sync

## Description

Install or upgrade ACS Operator from the Fleetshard-sync service using helm chart. The Operator upgrade runs before the Central instances reconciler loop.

It is implemented in the scope of Canary upgrade for Central instances feature. The next steps are:
- Run multiple ACS Operator version simultaneously
- Manage from Fleet Manager which Operator version should reconcile Central instance

## Checklist (Definition of Done)

~- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)~

## Test manual

First of all, the OLM based ACS Operator should be deleted without removing an existing CRDs
1. List and delete OLM subscription for the ACS Operator

```
kubectl get subscription -n stackrox-operator
kubectl delete subscription <subscription> -n stackrox-operator
```

2. Delete the Operator’s ClusterServiceVersion

```
kubectl get clusterserviceversion -n stackrox-operator
kubectl delete clusterserviceversion rhacs-operator.<version> -n stackrox-operator
```

3. Check that there is no running ACS Operator

```
kubectl get pods -n stackrox-operator
NAME                                                              READY   STATUS      RESTARTS      AGE
```

Then it's time to run and verify
4. Build and deploy

```
make deploy/dev
```

5.  Check that a new ACS Operator pod is running and neccessary CRDs exist

```
kubectl get pods -n stackrox-operator
NAME                                                              READY   STATUS      RESTARTS      AGE
rhacs-operator-controller-manager-66b6557d46-vfmbs                2/2     Running     1 (24h ago)   29h
...

kubectl describe crd securedclusters.platform.stackrox.io
Name:         securedclusters.platform.stackrox.io
Namespace:
Labels:       <none>
Annotations:  controller-gen.kubebuilder.io/version: v0.10.0
API Version:  apiextensions.k8s.io/v1
Kind:         CustomResourceDefinition
....

kubectl describe crd centrals.platform.stackrox.io
Name:         centrals.platform.stackrox.io
Namespace:
Labels:       <none>
Annotations:  controller-gen.kubebuilder.io/version: v0.10.0
API Version:  apiextensions.k8s.io/v1
Kind:         CustomResourceDefinition
...
```
